server/Makefile

```diff
@@ -457,17 +458,28 @@ modules-tidy:
 		mv channels/imports/imports.go.orig channels/imports/imports.go; \
 	fi;
 
-test-server-pre: check-prereqs-enterprise start-docker go-junit-report do-cover-file ## Runs tests.
+test-server-pre: check-prereqs-enterprise start-docker gotestsum ## Runs tests.
 ifeq ($(BUILD_ENTERPRISE_READY),true)
 	@echo Running all tests
 else
 	@echo Running only TE tests
 endif
 
+test-server-race: export MM_SERVER_PATH := $(MM_SERVER_PATH)
+test-server-race: export GOTESTSUM_FORMAT := $(GOTESTSUM_FORMAT)
+test-server-race: export GOTESTSUM_JUNITFILE := $(GOTESTSUM_JUNITFILE)
+test-server-race: export GOTESTSUM_JSONFILE := $(GOTESTSUM_JSONFILE)
```

Comment:
This is perhaps the ugliest part of these changes, as I have to duplicate them for every test. I opted not to clutter the command invocations themselves, and I discovered I can't do this in just `test-server-pre` as it only applies to the rule. There's other tricks like exporting all `Makefile` vars into the envrionment all the time, but I didn't want to risk unexpected breakages here.

Honestly, longer term, I'm wondering if we consider something like Mage or Taskfile to do this more cleanly anyway.